Description: build libs/silk before mod_silk
Author: <bdi@burnet.ru>

--- a/freeswitch/src/mod/codecs/mod_silk/Makefile	2010-03-13 15:43:08.000000000 +0800
+++ b/freeswitch/src/mod/codecs/mod_silk/Makefile	2010-03-13 20:56:01.000000000 +0800
@@ -1,8 +1,14 @@
 BASE=../../../..
 
-A=../../../../libs/silk/libSKP_SILK_SDK.a
+SILK_DIR=$(switch_srcdir)/libs/silk
+SILK_BUILDDIR=$(switch_builddir)/libs/silk
+SILKA=$(SILK_BUILDDIR)/libSKP_SILK_SDK.a
 
-LOCAL_INSERT_CFLAGS=if test -f $(A); then echo "-I../../../../libs/silk/interface" ; else echo "-DNOTFOUND" ; fi ;
-LOCAL_INSERT_LDFLAGS=test ! -f $(A) || echo $(A)
+LOCAL_CFLAGS=-I$(SILK_DIR)/interface
+LOCAL_LIBADD=$(SILKA)
 
 include $(BASE)/build/modmake.rules
+
+$(SILKA): $(SILK_DIR)/Makefile
+	cd $(SILK_BUILDDIR) && $(MAKE)
+	$(TOUCH_TARGET)
